channeldb/payments.go

```diff
@@ -116,6 +117,50 @@ func (db *DB) FetchAllPayments() ([]*OutgoingPayment, error) {
 	return payments, nil
 }
 
+// LookupPayment attempts to look up an payment according to its 32 byte
+// payment hash.
+func (db *DB) LookupPayment(paymentHash [32]byte) (*OutgoingPayment, error) {
+	var payment *OutgoingPayment
+
+	err := db.View(func(tx *bolt.Tx) error {
+		payments := tx.Bucket(paymentBucket)
+		if payments == nil {
+			return ErrNoPaymentsCreated
+		}
+
+		return payments.ForEach(func(k, v []byte) error {
+			// If the value is nil, then we ignore it as it may be
+			// a sub-bucket.
+			if v == nil {
+				return nil
+			}
+
+			r := bytes.NewReader(v)
```

Comment:
"r" could become "reader" or you could combine this line with the next one, removing the need for an intermediate variable.